src/core/initialize.rs

```diff
@@ -74,7 +83,25 @@ impl StackTraceGetter {
         Ok(self.get_trace_from_current_thread()?)
     }
 
-    fn get_trace_from_current_thread(&self) -> Result<StackTrace, MemoryCopyError> {
+    #[cfg(any(target_os = "macos", target_os = "windows", target_os = "linux"))]
+    fn is_on_cpu_os_specific(&self) -> Result<bool> {
+        // remoteprocess crate exposes a Thread.active() method for each of these targets
+        for thread in self.process.threads()?.iter() {
+            if thread.active()? {
+                return Ok(true);
+            }
+        }
+        Ok(false)
+    }
+
+    #[cfg(not(any(target_os = "macos", target_os = "windows", target_os = "linux")))]
+    fn is_on_cpu_os_specific(&self) -> Result<bool> {
+        /* We don't have OS specific checks for these targets,
+         * so fallback to using the interpreter based check down the line */
+        Ok(false)
```

Comment:
If you return `false` here and `on_cpu` is used, then nothing will be collected at all.

In this configuration - perhaps just deny using `on_cpu`?

Reply:
Nice catch, meant to return true 😅